woocommerce/woocommerce-ios
# Image Upload: media from camera capture and Photo Library

"Photo library" and "Camera capture" parts for #1713 

- [x] ⚠️ update PR base after https://github.com/woocommerce/woocommerce-ios/pull/1743 is merged ⚠️ 

## Summary

This PR presents the camera and device media library picker UI for the user to pick image(s).

## Changes

- Added `WPMediaPicker` pod
- In `CameraCaptureCoordinator`, presented the camera capture UI if the app has permissions for saving images to the Photo Library (otherwise the user would just lose the photo they just took) and camera
  - After an image is captured, saved the image to the Photo Library
  - Handled possible errors with localized description
  - Updated the permission copy for camera and Photo Library access - feel free to suggest other copy!
- In `DeviceMediaLibraryPicker`, presented `WPNavigationMediaPickerViewController` with `WPPHAssetDataSource` data source for picking images from the Photo Library
- In `ProductImagesViewController`, added some basic handling of camera capture and device media library picker completion

## Testing

- On a device, build the PR branch (otherwise the camera option wouldn't be shown)
- Go to Products tab
- Tap on a simple Product
- Tap on the "+" cell in the images header
- Tap "Add Photos" button
- Tap "Take a photo" --> should be prompted a permission alert for Photo Library and camera with clear reasons (feel free to suggest better copy!)
- Tap "Cancel" to exit the camera capture --> should be back to the Product images UI
- Tap "Add Photos" button and then "Take a photo" again
- Take a photo and use the photo --> should be back to the Product images UI
- Tap "Add Photos" button and then "Choose from device" --> should see the albums of the device Photo Library, and the photo you just took from camera
- Pick a few photos
- Tap "Add #" in the navigation bar --> should be back to the Product images UI
- Tap "Add Photos" button and then "Choose from device" again
- Pick a few photos
- Navigate back to the album screen
- Tap "Cancel" --> should be back to the Product images UI

Additionally, you can play with the camera and Photo Library permissions in the device settings and make sure appropriate alerts are shown.

## Example screenshots

Permission alert for Photo Library access | Permission alert for camera access | Device media library picker UI
-- | -- | --
![IMG_7654](https://user-images.githubusercontent.com/1945542/72606465-9c936780-3959-11ea-93ed-247d63293898.PNG) | ![IMG_7655](https://user-images.githubusercontent.com/1945542/72606467-9d2bfe00-3959-11ea-908b-3c49286fbbf4.PNG) | ![IMG_7658](https://user-images.githubusercontent.com/1945542/72606468-9d2bfe00-3959-11ea-8b20-ea0db92d6391.PNG)

Update release notes:
